syntax = "proto3";

package vacancy;

option go_package = "../protobuf";

// python3 -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. vacancies.proto

service Vacancies {
    rpc Create (CreateRequest) returns (Vacancy);
    rpc List (ListRequest) returns (ListResponse);
    rpc Get (GetRequest) returns (Vacancy);
    rpc Edit (Vacancy) returns (Vacancy);
}

message ListRequest{}

message Vacancy {
    int32  id = 1;
    int32  sect_id = 2;
    string name = 3;
    string description = 4;
    bool   is_active = 5;
    string token  = 6;
}


message ListResponse {
    repeated Vacancy vacancies = 1;
}

message GetRequest {
    int32 id = 1;
    string name = 2;
}

message CreateRequest {
    int32 sect_id = 1;
    string description  = 2;
}
